examples/ipfs-kad/src/main.rs

```diff
@@ -102,11 +121,42 @@ async fn main() -> Result<(), Box<dyn Error>> {
                         println!("Query timed out with no closest peers.");
                     }
                 }
-            };
+            }
+
+            break;
+        } else if let SwarmEvent::Behaviour(kad::Event::OutboundQueryProgressed {
```

Comment:
Can you rewrite this to a `match`? I think that would be more idiomatic than a chain of `if let`s!

Reply:
Done